src/extractor/guidance/turn_handler.cpp

```diff
@@ -22,6 +22,30 @@ namespace extractor
 namespace guidance
 {
 
+// @CHAU_TODO
+// a wrapper to handle road indices of forks at intersections
+struct TurnHandler::Fork
+{
+    const std::size_t right;
+    const std::size_t left;
+    const std::size_t size;
+    const bool valid;
```

Comment:
data + boolean valid is more or less what `optional<T>` is for - just make it a plain old `struct T { U data; };` and then use `optional<T>` and check with `if (myoptional) use(*myoptional);`